src/runtime-rs/crates/runtimes/common/src/types/mod.rs

```diff
@@ -0,0 +1,217 @@
+// Copyright (c) 2019-2022 Alibaba Cloud
+// Copyright (c) 2019-2022 Ant Group
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+mod trans_from_shim;
+mod trans_into_shim;
+
+use std::fmt;
+
+use anyhow::{Context, Result};
+use kata_sys_util::validate;
+use kata_types::mount::Mount;
+
+/// Request: request from shim
+/// Request and Response messages need to be paired
+#[derive(Debug, Clone)]
+pub enum Request {
+    CreateContainer(ContainerConfig),
+    CloseIo(ProcessID),
+    DeleteProcess(ProcessID),
+    Exec(ExecProcessRequest),
+    KillProcess(KillRequest),
+    Wait(ProcessID),
+    StartProcess(ProcessID),
+    StateProcess(ProcessID),
+    Shutdown(ShutdownRequest),
+    Pause(String),
+    Resume(String),
+    ResizePty(ResizePtyRequest),
+    Stats(String),
+    Update(UpdateRequest),
+    Pid,
+    Connect(String),
+}
+
+impl fmt::Display for Request {
+    fn fmt(&self, f: &mut std::fmt::Formatter) -> fmt::Result {
+        write!(f, "{:?}", &self)
+    }
+}
+
+/// Response: response to shim
+/// Request and Response messages need to be paired
+#[derive(Debug, Clone)]
+pub enum Response {
+    CreateContainer(ShimPid),
+    CloseIo,
+    DeleteProcess(ProcessStateInfo),
+    Exec,
+    KillProcess,
+    Wait(ProcessExitStatus),
+    StartProcess(ShimPid),
+    StateProcess(ProcessStateInfo),
+    Shutdown,
+    Pause,
+    Resume,
+    ResizePty,
+    StatsResponse(StatsInfo),
+    Update,
+    Pid(ShimPid),
+    Connect(ShimPid),
+}
+
+impl std::fmt::Display for Response {
+    fn fmt(&self, f: &mut std::fmt::Formatter) -> std::fmt::Result {
+        write!(f, "{:?}", &self)
+    }
+}
+
+#[derive(Clone, Copy, Debug, PartialEq)]
+pub enum ProcessType {
+    Container,
+    Exec,
+}
+
+#[derive(Clone, Debug)]
+pub struct ProcessID {
+    pub cid: String,
+    pub eid: String,
+    pub process_type: ProcessType,
+}
+
+impl fmt::Display for ProcessID {
+    fn fmt(&self, f: &mut std::fmt::Formatter) -> fmt::Result {
+        write!(f, "{:?}", &self)
+    }
+}
+
+impl ProcessID {
+    pub fn new(cid: &str, eid: &str) -> Result<Self> {
+        validate::verify_cid(cid).context("verify cid")?;
+        validate::verify_cid(eid).context("verify eid")?;
+        let (eid, process_type) = if eid.is_empty() || cid == eid {
+            ("".to_string(), ProcessType::Container)
+        } else {
+            (eid.to_string(), ProcessType::Exec)
+        };
+        Ok(Self {
+            cid: cid.to_string(),
+            eid,
+            process_type,
+        })
+    }
+}
+#[derive(Debug, Clone)]
+pub struct ContainerConfig {
+    pub cid: String,
+    pub bundle: String,
+    pub rootfs: Vec<Mount>,
```

Comment:
If it's a `Vec`, shouldn't that be called `mounts`?

Anyway, that does not match how it's done on the Go side, where there is a `RootFS` type:

```
// RootFs describes the container's rootfs.
type RootFs struct {
	// Source specifies the BlockDevice path
	Source string
	// Target specify where the rootfs is mounted if it has been mounted
	Target string
	// Type specifies the type of filesystem to mount.
	Type string
	// Options specifies zero or more fstab style mount options.
	Options []string
	// Mounted specifies whether the rootfs has be mounted or not
	Mounted bool
}
```

Reply:
Thanks for your review, Is it more appropriate to follow `shim.proto`, it defines as `rootfs: Vec<Mount>`, 
https://github.com/containerd/containerd/blob/main/runtime/v2/task/shim.proto#L57

Reply:
Thanks for your review, would it be clearer to rename it to `rootfs_mounts`.